torch/_functorch/_aot_autograd/jit_compile_runtime_wrappers.py

```diff
@@ -501,7 +501,11 @@ def aot_dispatch_autograd(
                     == MutationType.MUTATED_IN_GRAPH
                     and fw_metadata.input_info[i].mutates_storage_metadata
                 )
-                if bw_out is None and not metadata_mutation_in_graph:
+                is_non_leaf = (
```

Comment:
Another way to look at this one is `is_inside_autograd_graph/has_autograd_history = input.grad_fn is not None` (if you have that info here).
I would rename the variable either way as, as the condition shows, you are no using the standard "leaf" definition here.

Reply:
yes, agreed with the name change